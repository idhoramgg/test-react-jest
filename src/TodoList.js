import React from 'react'

const TodoList = (props) => {
    // 2
    let filteredTodos = props.todos;

    // 1
    let todoList = <li>No Item</li>
    
    // 3
    if(props.todos && props.todos.length > 0) {
        if(props.keyword) {
            filteredTodos = props.todos.filter(todo => (
                todo.toLowerCase().includes(props.keyword.toLowerCase())
            ))
        }
        todoList = filteredTodos.map((todo, i) => <li key={i}>{todo}</li>)
    }
    return <ul>{todoList}</ul>
}
export default TodoList;