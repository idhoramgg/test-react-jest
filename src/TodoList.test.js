import { render } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from './TodoList';

const todos = [
    "Satu",
    "Dua",
    "Tiga"
];

describe('todolist', () => {
    it('should render with no list', () => {
        let component = renderer.create(<TodoList />);
        expect(component).toMatchSnapshot();
    });

    it('should render with list', () => {
        let component = renderer.create(<TodoList todos={todos}/>);
        expect(component).toMatchSnapshot()
    })

    it('sholud render with filtered list', () => {
        let keywords = ['sa', 'u', 't'];
        keywords.forEach(keyword => {
            const component = renderer.create(
                <TodoList todos={todos} keyword={keyword} />
            );
            expect(component).toMatchSnapshot()
        })
    })
});