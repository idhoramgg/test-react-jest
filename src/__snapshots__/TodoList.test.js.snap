// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todolist sholud render with filtered list 1`] = `
<ul>
  <li>
    Satu
  </li>
</ul>
`;

exports[`todolist sholud render with filtered list 2`] = `
<ul>
  <li>
    Satu
  </li>
  <li>
    Dua
  </li>
</ul>
`;

exports[`todolist sholud render with filtered list 3`] = `
<ul>
  <li>
    Satu
  </li>
  <li>
    Tiga
  </li>
</ul>
`;

exports[`todolist should render with list 1`] = `
<ul>
  <li>
    Satu
  </li>
  <li>
    Dua
  </li>
  <li>
    Tiga
  </li>
</ul>
`;

exports[`todolist should render with no list 1`] = `
<ul>
  <li>
    No Item
  </li>
</ul>
`;
